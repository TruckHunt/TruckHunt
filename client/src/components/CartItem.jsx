import React from 'react';
import { Button, Label, Message, Grid, Header } from 'semantic-ui-react';
import utils from '../utils.js';

const CartItem = ({ name, price, quantity, onRemoveClicked }) => (
  <Message size="small" className="animated slideInUp">
    <Grid>
      <Grid.Row>
        <Grid.Column width={3} textAlign="center">
          <Label circular basic size="large" color="orange">{quantity}</Label>
        </Grid.Column>
        <Grid.Column width={10}>
          <Header as="h2">{name}</Header>
          <Label basic color="orange">&#36;{utils.formatCentsToDollars(price)}</Label>
        </Grid.Column>
        <Grid.Column width={3} textAlign="center">
          <Button circular color="orange" icon="minus" onClick={onRemoveClicked} size="small" />
        </Grid.Column>
      </Grid.Row>
    </Grid>
  </Message>
);

export default CartItem;
