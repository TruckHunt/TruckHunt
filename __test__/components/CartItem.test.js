import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import CartItem from '../../client/src/components/CartItem.jsx';

describe('CartItem Component Tests', () => {
  // snapshot test - to update, run "jest --updateSnapshot"
  // see https://facebook.github.io/jest/docs/snapshot-testing.html#content
  test('CartItem matches snapshot', () => {
    const component = renderer.create(<CartItem
      name="Burrito"
      price="699"
      quantity="1"
    />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  // test('CartItem has Burrito text', () => {
  //   const cartItem = mount(<CartItem />);
  //   expect(cartItem.text()).toEqual('Burrito');
  // });
});
