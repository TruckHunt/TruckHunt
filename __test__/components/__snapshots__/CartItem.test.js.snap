// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem Component Tests CartItem matches snapshot 1`] = `
<div
  className="ui small message animated slideInUp"
>
  <div
    className="ui grid"
  >
    <div
      className="row"
    >
      <div
        className="center aligned three wide column"
      >
        <div
          className="ui orange large basic circular label"
          onClick={[Function]}
        >
          1
        </div>
      </div>
      <div
        className="ten wide column"
      >
        <h2
          className="ui header"
        >
          Burrito
        </h2>
        <div
          className="ui orange basic label"
          onClick={[Function]}
        >
          $
          6.99
        </div>
      </div>
      <div
        className="center aligned three wide column"
      >
        <button
          className="ui orange small circular icon button"
          onClick={[Function]}
          tabIndex={undefined}
        >
          <i
            aria-hidden="true"
            className="minus icon"
          />
           
        </button>
      </div>
    </div>
  </div>
</div>
`;
